src/core/processing/models/qgsprocessingmodelalgorithm.cpp

```diff
@@ -769,6 +769,9 @@ void QgsProcessingModelAlgorithm::updateDestinationParameters()
         continue;
 
       std::unique_ptr< QgsProcessingParameterDefinition > param( source->clone() );
+      // Even if an output was hidden in a child algorithm, we want to show it here for the final
+      // outputs.
+      param->setFlags( param->flags() & ~QgsProcessingParameterDefinition::FlagHidden );
```

Comment:
Another very hacky-feeling thing to do. Does someone know what hidden parameters are for?